metasrv/tests/it/api/http/metrics.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_base::base::tokio;
+use databend_meta::api::http::v1::metrics::metrics_handler;
+use databend_meta::meta_service::MetaNode;
+use poem::get;
+use poem::http::Method;
+use poem::http::StatusCode;
+use poem::http::Uri;
+use poem::Endpoint;
+use poem::EndpointExt;
+use poem::Request;
+use poem::Route;
+use pretty_assertions::assert_eq;
+
+use crate::init_meta_ut;
+use crate::tests::service::MetaSrvTestContext;
+
+#[tokio::test]
+async fn test_metrics() -> common_exception::Result<()> {
+    let (_log_guards, ut_span) = init_meta_ut!();
+    let _ent = ut_span.enter();
+
+    let tc = MetaSrvTestContext::new(0);
+
+    let meta_node = MetaNode::start(&tc.config.raft_config).await?;
+
+    let cluster_router = Route::new()
+        .at("/v1/metrics", get(metrics_handler))
+        .data(meta_node.clone());
+    let response = cluster_router
+        .call(
+            Request::builder()
+                .uri(Uri::from_static("/v1/metrics"))
+                .method(Method::GET)
+                .finish(),
+        )
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), StatusCode::OK);
+
+    let body = response.into_body().into_vec().await.unwrap();
+    let metrics =
+        serde_json::from_str::<serde_json::Value>(String::from_utf8_lossy(&body).as_ref())?;
+    println!("{:?}", metrics);
+
+    let expected_json = "{
+        \"running_state\": {
+            \"Ok\": null
+        },
+        \"id\": 0,
+        \"state\": \"Leader\",
+        \"current_term\": 1,
+        \"last_log_index\": 2,
+        \"last_applied\": {
+            \"term\": 1,
+            \"index\": 2
+        },
+        \"current_leader\": 0,
```

Comment:
The values in a json are volatile.

The test needs to decode the json and check the keys one by one.
And for the values, only check the type of them.

Reply:
> The values in a json are volatile.
> 
> The test needs to decode the json and check the keys one by one. And for the values, only check the type of them.

OK. Waiting for confirming the content of the repsonse, or there will be a lot of useless work.